src/main/java/com/exasol/adapter/dialects/exasol/ExasolColumnMetadataReader.java

```diff
@@ -60,15 +60,36 @@ public DataType mapJdbcType(final JDBCTypeDescription jdbcTypeDescription) {
     @Override
     public JDBCTypeDescription readJdbcTypeDescription(final ResultSet remoteColumn) throws SQLException {
         final JDBCTypeDescription typeDescription = super.readJdbcTypeDescription(remoteColumn);
-        if (typeDescription.getJdbcType() == EXASOL_INTERVAL_DAY_TO_SECONDS) {
+        switch (typeDescription.getJdbcType()) {
+        case EXASOL_INTERVAL_DAY_TO_SECONDS:
             return extractIntervalDayToSecondPrecision(remoteColumn, typeDescription);
-        } else if (typeDescription.getJdbcType() == EXASOL_INTERVAL_YEAR_TO_MONTHS) {
+        case EXASOL_INTERVAL_YEAR_TO_MONTHS:
             return extractIntervalYearToMonthPrecision(remoteColumn, typeDescription);
-        } else {
+        case EXASOL_GEOMETRY:
+            return getGeometryWithExtractedSrid(remoteColumn, typeDescription);
+        default:
             return typeDescription;
         }
     }
 
+    private JDBCTypeDescription getGeometryWithExtractedSrid(final ResultSet remoteColumn,
+            final JDBCTypeDescription typeDescription) throws SQLException {
+        final String typeDescriptionString = getTypeDescriptionStringForColumn(remoteColumn);
+        final int srid = extractSrid(typeDescriptionString);
+        return new JDBCTypeDescription(typeDescription.getJdbcType(), typeDescription.getDecimalScale(), srid,
+                typeDescription.getByteSize(), typeDescription.getTypeName());
+    }
+
+    private int extractSrid(final String typeDescriptionString) {
+        final int start = typeDescriptionString.indexOf("(");
+        if (start > -1) {
+            final int end = typeDescriptionString.indexOf(")");
+            return Integer.parseInt(typeDescriptionString.substring(start + 1, end));
+        } else {
+            return DEFAULT_SPACIAL_REFERENCE_SYSTEM_IDENTIFIER;
+        }
```

Comment:
Time to practice regular expressions. Please replace by a regex with a capture group for the contents of the brackets.

Mind that Sonar does not like + / *, so use `{n,m}` with resonable `m` boundary instead.

Reply:
Challenge accepted